src/python/pants/util/meta.py

```diff
@@ -6,6 +6,13 @@
 from typing import Any, Callable, Optional, Type, TypeVar, Union
 
 
+T = TypeVar("T")
+C = TypeVar("C")
+# This TypeVar is used to annotate property decorators. The decorator should take the parameter
+# func=Callable[..., P] and return P.
+P = TypeVar("P")
```

Comment:
Given that we're importing this from another file, is it possible to give it a more descriptive name? Is it possible to have an associated docstring for `TypeVar`s? I'll check that out!

Reply:
I realized that we should actually just use `T`. There is nothing special about `P`! Using `T` vs. `P` suggests there is special magic associated with `P`, when really it's just a normal type var.